### PR TITLE
Clarify how workflow can be injected

### PR DESCRIPTION
**Pull request by epitre**

### CODE REVIEW

File: components/workflow.rst [hunk 1]
```diff
@@ -84,7 +84,7 @@ you can retrieve a workflow from it and use it as follows::
     // ...
     // Consider that $blogPost is in place "draft" by default
     $blogPost = new BlogPost();
-    $workflow = $registry->get($blogPost);
+    $workflow = $registry->get(BlogPost::class);
```
Comment (noniagriconomie) [inline components/workflow.rst:87]: you can not change this like that as a previous version of sf will break
you must add a versionadded directive instead documentation this new way available :)

**Comment by epitre** [comment]
@lyrixx good for you ?

File: workflow.rst [hunk 2]
```diff
@@ -265,6 +265,36 @@ registry in the constructor::
         }
     }
 
+Or use the typed + named parameter injection::
```
Comment (lyrixx) [inline workflow.rst:268]: I would put this paragraph before the one about the registry.

File: workflow.rst [hunk 3]
```diff
@@ -265,6 +265,36 @@ registry in the constructor::
...
+
+    use App\Entity\BlogPost;
+    use Symfony\Component\Workflow\WorkflowInterface;
+
+    class MyClass
+    {
+        private $blogPostWorkflow;
+
+        public function __construct(WorkflowInterface $blogPostWorkflow)
```
Comment (lyrixx) [inline workflow.rst:277]: This looks wrong to me, it should be $blogPublishingWorkflow according to the definition https://github.com/symfony/symfony-docs/blob/018b64c26fa07591c551da0619058918899ddc61/workflow.rst#creating-a-workflow

File: workflow.rst [hunk 4]
```diff
@@ -828,25 +858,23 @@ Then you can access this metadata in your controller as follows::
 
     // src/App/Controller/BlogPostController.php
     use App\Entity\BlogPost;
-    use Symfony\Component\Workflow\Registry;
+    use Symfony\Component\Workflow\WorkflowInterface;
     // ...
 
-    public function myAction(Registry $registry, BlogPost $post)
+    public function myAction(WorkflowInterface $blogPostWorkflow, BlogPost $post)
```
Comment (lyrixx) [inline workflow.rst:864]: Same here

**Review by lyrixx — COMMENTED** [review]
Thanks for your PR. I felt a couple of comment.
Could you also change the PR title, because it does not reflect the PR anymore (the title is used to create the commit message)

File: workflow.rst (hunk 4 above)
Comment (epitre) [reply workflow.rst:864]: :heavy_check_mark:

File: workflow.rst (hunk 3 above)
Comment (epitre) [reply workflow.rst:277]: :heavy_check_mark:

File: workflow.rst (hunk 2 above)
Comment (epitre) [reply workflow.rst:268]: :heavy_check_mark:

**Review by lyrixx — APPROVED** [review]
👍🏼

**Review by Nyholm — APPROVED** [review]
Thank you @epitre 

This is great!

**Comment by wouterj** [comment]
Thank you @epitre for your research on this topic and the time you put into sharing your findings with others! This PR is perfect :)

**Comment by wouterj** [comment]
Fyi, I've merged this PR in the 4.4 version of the docs and I'll bring it in newer version from there. This is why I cannot mark this PR as "merged". But here is the proof that your first doc contribution is a fact: 4cb6b21 :)

**Pull request closed without merging**